### ISSUE TITLE
Support Fuyu-8B

### ISSUE THREAD

**Issue by dongxiaolong** [axolotl-ai-cloud/axolotl#777]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/OpenAccess-AI-Collective/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/OpenAccess-AI-Collective/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

[Fuyu-8B](https://www.adept.ai/blog/fuyu-8b) is a multi-modal text and image transformer trained by [Adept AI](https://www.adept.ai/).

Architecturally, Fuyu is a vanilla decoder-only transformer - there is no image encoder. Image patches are instead linearly projected into the first layer of the transformer, bypassing the embedding lookup. We simply treat the transformer decoder like an image transformer (albeit with no pooling and causal attention). See the below diagram for more details.

### ✔️ Solution

Multimodal training support.

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by winglian** [comment]
Hi @dongxiaolong, Fuyu is interesting in that we can simply feed chunks of arbitrarily sized images into the model. I also have multimodal using Llava on the roadmap, but if we can understand better how Fuyu works, it might be feasible to plan to architect multimodal training for both methodologies. Is this something you'd be interested in helping to do?

**Comment by dongxiaolong** [comment]
> Hi @dongxiaolong, Fuyu is interesting in that we can simply feed chunks of arbitrarily sized images into the model. I also have multimodal using Llava on the roadmap, but if we can understand better how Fuyu works, it might be feasible to plan to architect multimodal training for both methodologies. Is this something you'd be interested in helping to do?

Hi @winglian,

I noticed the [Fuyu Finetuning Example](https://github.com/huggingface/transformers/pull/26997). It appears that the fine-tuning details for Fuyu are currently being developed under the transformers library. You can find more information in the [transformers' Fuyu directory](https://github.com/huggingface/transformers/tree/main/src/transformers/models/fuyu).

As of now, I'm not aware of any additional details regarding the model's training that have been shared publicly. It might be beneficial to reach out to the original authors for further insights.

If there's any way I can assist, please let me know. I have access to two A100 GPUs that could be utilized for this purpose.

**Comment by Stillerman** [comment]
I am interested in assisting in this

**Comment by NanoCode012** [comment]
With the addition of the Fuyu to transformers, axolotl should inherently support it. `sample_packing` and `flash_attention` would not work as their modeling code does not support it.
